### PR TITLE
fix(positions): detect and repair duplicate task positions automatically

### PR DESCRIPTION
**Pull request by kolaente**
## Summary

- Add automatic conflict detection to task position updates that redistributes duplicate positions within available gaps
- Add CLI command `vikunja repair-task-positions` with `--dry-run` flag for bulk repair across all views
- Update frontend `calculateItemPosition()` to handle edge case where neighbors have identical positions

## Background

Relates to: https://community.vikunja.io/t/reordering-not-possible-position-value-the-same-for-different-tasks/4078

Duplicate positions can occur due to race conditions or historical bugs, causing tasks to appear in the wrong order or jump around when the page is refreshed.

## Changes

**Backend:**
- `ErrNeedsFullRecalculation` sentinel error and `MinPositionSpacing` constant (1e-9)
- `findPositionConflicts()` - detects duplicate positions in a view
- `resolveTaskPositionConflicts()` - redistributes conflicting tasks within available gaps
- `updateTaskPosition()` - now detects and resolves conflicts after writes
- `RepairTaskPositions()` - bulk repair function for CLI
- New CLI command `vikunja repair-task-positions --dry-run`

**Frontend:**
- `calculateItemPosition()` handles equal neighbor positions by nudging above
- `MIN_POSITION_SPACING` constant matching backend

## Test Plan

- [x] Backend unit tests for conflict detection and resolution
- [x] Frontend unit tests for calculateItemPosition edge cases
- [x] Build passes
- [x] Lint passes (0 issues)
- [ ] Manual testing: drag tasks in kanban, verify order persists after refresh
- [ ] Manual testing: run `vikunja repair-task-positions --dry-run` on existing DB

## Summary by CodeRabbit

* **New Features**
  * Added a `repair-task-positions` CLI command to detect and resolve task position conflicts, with dry-run preview option.
  * Implemented automatic conflict detection and resolution to ensure unique task positions.

* **Bug Fixes**
  * Fixed position calculation edge cases and JSON serialization precision issues.
  * Improved handling of conflicting task positions with deterministic resolution.

* **Tests**
  * Added comprehensive test coverage for position repair and conflict resolution logic.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by kolaente** [comment]
auggie review

**Comment by kolaente** [comment]
@codex review

File: pkg/models/task_position.go [hunk 1]
```diff
@@ -80,22 +92,61 @@ func updateTaskPosition(s *xorm.Session, a web.Auth, tp *TaskPosition) (err erro
...
 		}
-		if shouldRecalculate {
-			return RecalculateTaskPositions(s, view, a)
+	} else {
+		_, err = s.
+			Where("task_id = ? AND project_view_id = ?", tp.TaskID, tp.ProjectViewID).
+			Cols("project_view_id", "position").
+			Update(tp)
+		if err != nil {
+			return
```
Comment (kolaente) [inline pkg/models/task_position.go:94]: This whole pattern could be refactored to `update on conflict do insert` at some point.

File: pkg/models/task_position.go [hunk 2]
```diff
@@ -80,22 +92,61 @@ func updateTaskPosition(s *xorm.Session, a web.Auth, tp *TaskPosition) (err erro
...
-		Cols("project_view_id", "position").
-		Update(tp)
+	// Check for and resolve position conflicts
+	conflicts, err := findPositionConflicts(s, tp.ProjectViewID, tp.Position)
 	if err != nil {
-		return
+		return err
+	}
+
+	if len(conflicts) > 1 {
```
Comment (kolaente) [inline pkg/models/task_position.go:117]: If we're recalculating anyway, we don't need to check for conflicts

File: pkg/models/task_position.go [hunk 3]
```diff
@@ -80,22 +92,61 @@ func updateTaskPosition(s *xorm.Session, a web.Auth, tp *TaskPosition) (err erro
...
+		err = resolveTaskPositionConflicts(s, tp.ProjectViewID, conflicts)
+		if errors.Is(err, ErrNeedsFullRecalculation) {
+			return RecalculateTaskPositions(s, view, a)
+		}
+		if err != nil {
+			return err
+		}
+
+		// Refresh tp.Position from DB so the API response reflects the actual stored value
+		updatedPosition := &TaskPosition{}
```
Comment (kolaente) [inline pkg/models/task_position.go]: move re-fetching to after the `if shouldRecalculate` so that it's only required once

File: pkg/models/task_position.go [hunk 4]
```diff
@@ -312,3 +409,219 @@ func createPositionsForTasksInView(s *xorm.Session, tasks []*Task, view *Project
...
+	result := &RepairResult{}
+
+	// Get all project view IDs that have task positions
+	var allPositions []*TaskPosition
+	err := s.GroupBy("project_view_id").Find(&allPositions)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pos := range allPositions {
```
Comment (kolaente) [inline pkg/models/task_position.go:432]: Add a step earlier that transforms the slice into a map that's keyd by view id, then loop over that to detect and fix the duplicate positions

**Comment by kolaente** [comment]
@coderabbitai full review

**Comment by kolaente** [comment]
@codex review

**Comment by kolaente** [comment]
auggie review
